src/__init__.py

```diff
@@ -84,6 +84,35 @@ def run_reconcile(clients: Clients, user_inputs: UserInputs) -> ReconcileOutputs
 
     # tee creates a copy of the iterator which is needed as check.conflicts consumes the iterator
     # it is passed
+    actions, check_actions = tee(actions, 2)
+    if reconcile.is_same_content(index, check_actions):
```

Comment:
As discussed in mattermost, I think we just need to change the check on line 113 of `__init__.py` be something like `not user_inputs.dry_run or <all actions noop>` and that should get us what we need - any update action should be considered as a difference event if the content doesn't change because the navigation table would be changing in that case

Reply:
I have reverted the business logic to only execute if there are NoopActions, instead of also considering UpdateAction with no content change, that, as discussed, may mean a different in the tree structure

However I believe it is actually sane to leave the check and the business logic where it is. We would nevertheless need to `tee` the iterator, and also having checks there can prevent the code execution after which is totally useless if there are only NoopAction. Beside this should apply regardless if the action is running in dry mode or not